devicetypes/rachio/rachio-iro2-controller.src/rachio-iro2-controller.groovy

```diff
@@ -0,0 +1,674 @@
+            state("default", label: 'Last Watered:\n${currentValue}')
+        }
+        standardTile("leftZoneTimeButton", "device.curZoneWaterTime", inactiveLabel: false, decoration: "flat") {
+            state "default", action:"decZoneWaterTime", icon:"st.thermostat.thermostat-left"
+        }
+        valueTile("curZoneWaterTime", "device.curZoneWaterTime", width: 2, height: 1, decoration: "flat") {
+            state "default", label:'Manual Zone Time:\n${currentValue} Minutes'
+        }
+        standardTile("rightZoneTimeButton", "device.curZoneWaterTime", inactiveLabel: false, decoration: "flat") {
+            state "default", action:"incZoneWaterTime", icon:"st.thermostat.thermostat-right"
+        }
+        valueTile("runAllZonesTile", "device.curZoneWaterTime", inactiveLabel: false, width: 2 , height: 1, decoration: "flat") {
+            state("default", label: 'Run All Zones\n${currentValue} Minutes', action:'runAllZones')
+        }
+        standardTile("standbyMode", "device.standbyMode", decoration: "flat", wordWrap: true, width: 2, height: 2) {
+            state "on", label:'Turn Standby Off', action:"standbyOff", nextState: "false", icon: "http://cdn.device-icons.smartthings.com/sonos/play-icon@2x.png"
+            state "off", label:'Turn Standby On', action:"standbyOn", nextState: "true", icon: "http://cdn.device-icons.smartthings.com/sonos/pause-icon@2x.png"
+        }
+        standardTile("refresh", "device.power", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+            state "default", label:'', action:"refresh.refresh", icon:"st.secondary.refresh"
+        }
+	}
+	main "valveTile"
+	details(["valveTile", "hardwareModel", "hardwareDesc", "activeZoneCnt", "curZoneIsCyclingTile", "leftButtonControl", "rainDelay", "rightButtonControl", "applyRainDelay",
+            "leftZoneTimeButton", "curZoneWaterTime", "rightZoneTimeButton", "runAllZonesTile", "lastUpdatedDt", "standbyMode", "refresh"])
+}
+
+def getAppImg(imgName)	{ return "https://raw.githubusercontent.com/tonesto7/rachio-manager/master/images/$imgName" }
+
+// parse events into attributes
+def parse(String description) {
+	log.debug "Parsing '${description}'"
+}
+
+def initialize() {
+	sendEvent(name: "DeviceWatch-DeviceStatus", value: "online", displayed: false, isStateChange: true)
+    sendEvent(name: "DeviceWatch-Enroll", value: groovy.json.JsonOutput.toJson(["protocol":"cloud", "scheme":"untracked"]), displayed: false)
+    
+    verifyDataAttr()
+}
+
+def verifyDataAttr() {
+	updateDataValue("HealthEnrolled", "true")
+    updateDataValue("manufacturer", "Rachio")
+    def gen = state?.deviceId ? parent?.getDevGeneration(state?.deviceId) : null
+    updateDataValue("model", "${device?.name}${gen ? " ($gen)" : ""}")
+}
+
+void installed() {
+	initialize()
+	state.isInstalled = true
+}
+
+void updated() {
+	initialize()
+}
+
+def ping() {
```

Comment:
ping() is only called for tracked devices, this DTH is enrolled as "untracked" thus ping is not called and should be removed